cpp/util/StreamSplitter.cpp

```diff
@@ -38,13 +42,18 @@ Snowflake::Client::Util::StreamSplitter::StreamSplitter(
 }
 
 Snowflake::Client::Util::ByteArrayStreamBuf*
-Snowflake::Client::Util::StreamSplitter::FillAndGetBuf(int bufIndex)
+Snowflake::Client::Util::StreamSplitter::FillAndGetBuf(
+  int bufIndex, int &partIndex)
 {
   _mutex_lock(&streamMutex);
-  m_inputStream->read(buffers[bufIndex]->getDataBuffer(), m_partMaxSize);
-  buffers[bufIndex]->updateSize(m_inputStream->gcount());
+  ByteArrayStreamBuf * buf = buffers[bufIndex];
+  memset(buf->getDataBuffer(), 0, m_partMaxSize);
+  m_inputStream->read(buf->getDataBuffer(), m_partMaxSize);
+  buf->updateSize(m_inputStream->gcount());
+  m_currentPartIndex ++;
+  partIndex = m_currentPartIndex;
```

Comment:
Basically this is the fix. Previously, this FillAndGetBuf is called by multiple tasks concurrently to fill the buffer with data and upload.However, the part index is not returned to the task. So the task use its taskId as upload partId, which could lead to data is not in right order. I am surprised that it failed rarely.